Mathlib/Topology/Algebra/TopologicallyNilpotent.lean

```diff
@@ -0,0 +1,130 @@
+    IsTopologicallyNilpotent (φ a) := by
+  unfold IsTopologicallyNilpotent at ha ⊢
+  simp_rw [← map_pow]
+  exact (map_zero φ ▸ hφ.tendsto 0).comp ha
+
+/-- `0` is topologically nilpotent -/
+theorem zero : IsTopologicallyNilpotent (0 : R) :=
+  tendsto_atTop_of_eventually_const (i₀ := 1)
+    (fun _ hi => by rw [zero_pow (Nat.ne_zero_iff_zero_lt.mpr hi)])
+
+theorem exists_pow_mem_of_mem_nhds {a : R} (ha : IsTopologicallyNilpotent a)
+    {v : Set R} (hv : v ∈ 𝓝 0) :
+    ∃ n, a ^ n ∈ v :=
+  (ha.eventually_mem hv).exists
+
+end MonoidWithZero
+
+section Ring
+
+variable {R : Type*} [TopologicalSpace R] [Ring R]
+
+/-- If `a` and `b` commute and `a` is topologically nilpotent,
+  then `a * b` is topologically nilpotent. -/
+theorem mul_right_of_commute [IsLinearTopology Rᵐᵒᵖ R]
+    {a b : R} (ha : IsTopologicallyNilpotent a) (hab : Commute a b) :
+    IsTopologicallyNilpotent (a * b) := by
+  simp_rw [IsTopologicallyNilpotent, hab.mul_pow]
+  exact IsLinearTopology.tendsto_mul_zero_of_left _ _ ha
+
+/-- If `a` and `b` commute and `b` is topologically nilpotent,
+  then `a * b` is topologically nilpotent. -/
+ theorem mul_left_of_commute [IsLinearTopology R R] {a b : R}
+    (hb : IsTopologicallyNilpotent b) (hab : Commute a b) :
+    IsTopologicallyNilpotent (a * b) := by
+  simp_rw [IsTopologicallyNilpotent, hab.mul_pow]
+  exact IsLinearTopology.tendsto_mul_zero_of_right _ _ hb
+
+/-- If `a` and `b` are topologically nilpotent and commute,
+  then `a + b` is topologically nilpotent. -/
+theorem add_of_commute [IsLinearTopology R R] {a b : R}
+    (ha : IsTopologicallyNilpotent a) (hb : IsTopologicallyNilpotent b) (h : Commute a b) :
+    IsTopologicallyNilpotent (a + b) := by
+  simp only [IsTopologicallyNilpotent, atTop_basis.tendsto_iff IsLinearTopology.hasBasis_ideal,
+    true_and]
+  intro I I_mem_nhds
+  obtain ⟨na, ha⟩ := ha.exists_pow_mem_of_mem_nhds I_mem_nhds
+  obtain ⟨nb, hb⟩ := hb.exists_pow_mem_of_mem_nhds I_mem_nhds
+  exact ⟨na + nb, fun m hm ↦
+    I.add_pow_mem_of_pow_mem_of_le_of_commute ha hb (le_trans hm (Nat.le_add_right _ _)) h⟩
+
+end Ring
+
+section CommRing
+
+variable {R : Type*} [TopologicalSpace R] [CommRing R] [IsLinearTopology R R]
+
+/-- If `a` is topologically nilpotent, then `a * b` is topologically nilpotent. -/
+theorem mul_right {a : R} (ha : IsTopologicallyNilpotent a) (b : R) :
+    IsTopologicallyNilpotent (a * b) :=
+  ha.mul_right_of_commute (Commute.all _ _)
+
+/-- If `b` is topologically nilpotent, then `a * b` is topologically nilpotent. -/
+ theorem mul_left (a : R) {b : R} (hb : IsTopologicallyNilpotent b) :
+    IsTopologicallyNilpotent (a * b) :=
+  hb.mul_left_of_commute (Commute.all a b)
+
+/-- If `a` and `b` are topologically nilpotent, then `a + b` is topologically nilpotent. -/
+theorem add {a b : R} (ha : IsTopologicallyNilpotent a) (hb : IsTopologicallyNilpotent b) :
+    IsTopologicallyNilpotent (a + b) :=
+  ha.add_of_commute hb (Commute.all _ _)
```

Comment:
```suggestion
  ha.add_of_commute hb (Commute.all ..)
```

Reply:
For some reason it seems this one did not get included